msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -168,6 +168,15 @@ private AuthenticationResultSupplier getAuthenticationResultSupplier(MsalRequest
         return supplier;
     }
 
+    private IHttpClient determineHttpClientType(Builder<?> builder) {
+
+        if (this instanceof ManagedIdentityApplication) {
```

Comment:
This is an inheritance warning sign. Base class should not be aware of its implementations.

Reply:
Agreed, I just did it for simplicity and to avoid (nearly) duplicate code, but it's definitely a bad practice. 

In the latest commit I did a proper refactor, and moved the ServiceBundle creation down into the implementation classes so these sorts of 'instanceof' checks aren't needed.